quantum/split_common/matrix.c

```diff
@@ -160,19 +167,21 @@ static bool read_rows_on_col(matrix_row_t current_matrix[], uint8_t current_col)
     for (uint8_t row_index = 0; row_index < ROWS_PER_HAND; row_index++) {
         // Store last value of row prior to reading
         matrix_row_t last_row_value = current_matrix[row_index];
+        matrix_row_t current_row_value = last_row_value;
```

Comment:
Why do we use the last row value as the starting point, instead of just starting with zero?
We can remove the entire `else` branch if we do, right?

Reply:
Its due to the matrix being an array of rows. And we have to do the unset logic as we scan. Where 'S' is the selected column, we are basically manipulating...

```text
     S
0b00|0|00000
0b00|0|00000
0b00|0|00000
0b00|0|00000

then in another iteration of read_rows_on_col

      S
0b000|0|0000
0b000|0|0000
0b000|0|0000
0b000|0|0000
```

I have been tinkering with removing row2col and bodging what #8930 did at runtime, but this is way more effort.